geode-management/src/main/java/org/apache/geode/management/internal/api/ClusterManagementResult.java

```diff
@@ -17,30 +17,71 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import com.fasterxml.jackson.annotation.JsonIgnore;
-
-
 public class ClusterManagementResult {
   private Map<String, Status> memberStatuses = new HashMap<>();
 
   private Status persistenceStatus = new Status(Status.Result.NOT_APPLICABLE, null);
 
-  public ClusterManagementResult() {}
+  private boolean successfullyAppliedOnMembers = false;
+
+  private boolean successfullyPersisted = false;
+
+  private boolean successful = false;
+
+  public ClusterManagementResult() {
+    generateSuccessStatus();
+  }
 
   public ClusterManagementResult(boolean success, String message) {
     this.persistenceStatus = new Status(success, message);
+
+    generateSuccessStatus();
+  }
+
+  public ClusterManagementResult(Status.Result status, String message) {
+    this.persistenceStatus = new Status(status, message);
+
+    generateSuccessStatus();
+  }
+
+  private void generateSuccessStatus() {
```

Comment:
What are your thoughts on creating a `ClusterManagementResultFactory`. The business logic below can be built into the factory and tested in isolation. This would allow the `ClusterManagementResult` to be somewhat static and simply jackson's mapping to json